sdk-lib/src/main/rust/utils.rs

```diff
@@ -77,3 +89,75 @@ where
     }
     outer
 }
+
+pub(crate) fn p2pkh_full_viewing_key<P: consensus::Parameters>(
+    params: &P,
+    seed: &[u8],
+    account: AccountId,
+) -> Result<ExtendedPubKey, hdwallet::error::Error> {
+    let pk = ExtendedPrivKey::with_seed(&seed)?;
+    let private_key = pk
+        .derive_private_key(KeyIndex::hardened_from_normalize_index(44)?)?
+        .derive_private_key(KeyIndex::hardened_from_normalize_index(params.coin_type())?)?
+        .derive_private_key(KeyIndex::hardened_from_normalize_index(account.0)?)?;
+    Ok(ExtendedPubKey::from_private_key(&private_key))
+}
+
+pub(crate) fn p2pkh_addr(
+    fvk: ExtendedPubKey,
+    index: DiversifierIndex,
+) -> Result<TransparentAddress, hdwallet::error::Error> {
+    let pubkey = fvk
+        .derive_public_key(KeyIndex::Normal(0))?
+        .derive_public_key(KeyIndex::Normal(u32::from_le_bytes(
+            index.0[..4].try_into().unwrap(),
+        )))?
+        .public_key;
+    Ok(derive_transparent_address_from_public_key(&pubkey))
+}
+
+/// This is temporary, and will be replaced by `zcash_address::unified::Ufvk`.
+pub(crate) fn fake_ufvk_encode(p2pkh: &ExtendedPubKey, sapling: &ExtendedFullViewingKey) -> String {
```

Comment:
Since zcash_address doesn't have any dependency upon zcash_primitives, why not just pull that in now? The `zcash_address` crate didn't even exist at the time of `zcash_primitives = "0.5"` so there isn't any conflict.

Reply:
I did try this, but none of the `zcash_primitives::zip32` refactors are present in this branch, so there's a mismatch between the types I have at my disposal, and the data that `zcash_address::unified` allows to be serialized. I figured this is only going to exist for a single PR, and then will disappear, so it's a helpful stepping stone.

Reply:
I was able to jump all the way to proper UFVK encodings in this PR, so this is now just an internal detail of the branch.